src/sinks/prometheus/collector.rs

```diff
@@ -298,6 +300,12 @@ impl TimeSeries {
         labels.sort();
         labels
     }
+
+    fn now(&mut self) -> i64 {
+        *self
+            .now
+            .get_or_insert_with(|| Utc::now().timestamp_millis())
```

Comment:
The naming of this function wouldn't indicate to me that it caches the value of `now`. I'm wondering if we should just remove that caching.

Alternatively, I could see making this a top-level function and using `cfg(test)` to define a mock implementation with a known value that the tests could depend on here.

Reply:
I figured it being both a simple optimization and a way of providing that all metrics in a batch used the same default timestamp. It has to be per-batch, as otherwise all metrics without a timestamp would share the same time again.

Reply:
That said, I am definitely open to renaming it to indicate the purpose better. I'll figure a more descriptive name.

Reply:
I can see why it might be desirable for all metrics in the same batch to have the same default timestamp value, but is that an actual requirement? I was thinking it could just always use whatever `Utc::now()` is, without the caching.

Reply:
👍 the naming is definitely better now though.

Reply:
No, it's not a requirement, just an optimization. On my desktop, `Utc::now` is 4-6x as expensive as cloning a string, though amusingly it's *faster* than cloning on my MBP (which hints that MacOS has an *aweful* malloc implementation).